treadmill-aws/sbin/start_openldap.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
+
+set -e
+
+if [ -z ${TREADMILL_INSTALL_DIR} ]; then
+    TREADMILL_INSTALL_DIR=/var/lib/treadmill-openldap
+fi    
+
+if [ -z ${DISTRO} ]; then
+    DISTRO=$(cd $(dirname $0)/.. && pwd)
+fi
+
+INSTALL_DIR=${TREADMILL_INSTALL_DIR}
+unset TREADMILL_INSTALL_DIR
+unset TREADMILL_ISA
+
+function openldap_usage {
+    MSG=$1
+    if [ "${MSG}" != "" ]; then
+        echo Usage error:
+        echo 
+        echo "  ${MSG}"
+        echo
+    fi
+    cat << USAGE
+Usage: 
+    $0 [OPTIONS]
+
+Options:
+   -m master1,master2            : List of openldap masters
+   -h                            : Print help and exit
+USAGE
+
+    exit 1
+}
+
+OPTIND=1
+while getopts "m:" OPT; do
+    case "${OPT}" in
+        m)
+            TREADMILL_LDAP=${OPTARG}
+            ;;
+        h)
+            openldap_usage
+            ;;
+        *)
+            openldap_usage
+            ;;
+    esac
+done
+shift $((OPTIND-1))
+
+echo "Installing Treadmill OpenLDAP: ${INSTALL_DIR}"
+
+/bin/mount --make-rprivate /
+
+HOSTNAME=$(hostname --fqdn)
+export HOSTNAME
+
+# Repeat kinit until successful
+until kinit -k -l 1d;
+do
+    echo Sleeping until server joined to IPA...
+    sleep 10
+done
+
+klist
+
+export TREADMILL_KRB_REALM=$(k-realm)
+export KRB5_KTNAME=/var/spool/keytabs-services/ldap-${HOSTNAME}.keytab
+
+mkdir -pv ${INSTALL_DIR}
+mkdir -p /var/spool/keytabs-services/
+
+response=$(ipa service-add ldap/${HOSTNAME}@${TREADMILL_KRB_REALM} 2>&1)
+RESULT=$?
+if [ ${RESULT} -eq 0 ]; then
+	ipa-getkeytab -p ldap/${HOSTNAME}@${TREADMILL_KRB_REALM} -k ${KRB5_KTNAME}
+else
+	echo ${response} | grep -q 'already exists' && echo 'Service already exists' || echo ${response}
+fi
+
+chown treadmld:treadmld ${KRB5_KTNAME}
```

Comment:
treadmld can't be hardcoded... must be parameter.

Reply:
Fixed.